platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolutionHost.java

```diff
@@ -17,17 +17,25 @@
 package org.gradle.api.internal.artifacts.configurations;
 
 import org.gradle.api.artifacts.ResolveException;
+import org.gradle.api.problems.internal.InternalProblems;
 import org.gradle.internal.Describables;
 import org.gradle.internal.DisplayName;
+import org.gradle.internal.component.resolution.failure.ReportableAsProblem;
+import org.gradle.internal.exceptions.MultiCauseException;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
 
 /**
  * The "Host" or owner of a resolution -- the thing in charge of the resolution, or the thing being resolved.
  *
  * <p>The purpose of this type is to be a configuration-cache compatible representation of the thing
  * being resolved. This type should remain as minimal as possible.</p>
+ *
+ * TODO: Split the interface into two: one for tracking what is doing the resolving and one for mapping resolution problems
```

Comment:
💭  I mean in truth we already have `ResolveExceptionMapper` which most implementations of this interface delegate to.

Reply:
It looks like we could use that, we'd just have to refactor to pass it alongside the host everywhere, then we could call it directly.

Reply:
I added a note about this potential refactoring